cpp/docs/DEVELOPER_GUIDE.md

```diff
@@ -362,14 +363,14 @@ void external_function(...);
 
 // cpp/include/cudf/detail/header.hpp
 namespace detail{
-void external_function(..., rmm::cuda_stream_view stream = rmm::cuda_stream_default)
+void external_function(..., rmm::cuda_stream_view stream)
 } // namespace detail
 
 // cudf/src/implementation.cpp
 namespace detail{
-    // defaulted stream parameter
+    // use stream parameter in detail implementation
     void external_function(..., rmm::cuda_stream_view stream){
-        // implementation uses stream w/ async APIs
+        // implementation uses stream with async APIs
```

Comment:
Sorry that I'm a bit OCD about comments: First letter capitalized, and punctuation at the end.
```suggestion
        // Implementation uses stream with async APIs.
```